RIOT-OS/RIOT
# cpu/esp32/freertos: fix semaphore take

### Contribution description


Involved parts: `cpu/esp32/freertos/semphr.c` and `tests/freertos`

Added test to test freertos. At the moment only tests for `esp32-wroom-32` board the mutex semaphore creation and `xSemaphoreTake()` and the reucursive mutex semaphore creation and `xSemaphoreTakeRecursive()`.
The future goal for this test is to test the complete freertos compatibility layer and to have the test for a general freertos compatibility layer that does not only work for esp32. I am currently working on a general freertos compatibility layer and trying to implement a test for it first, while doing it i noticed a bug.

Bug fix for `cpu/esp32/freertos/semphr.c::xSemaphoreTakeRecursive()` and c`pu/esp32/freertos/semphr.c::xSemaphoreTake()` return value when `xTicksToWait `is 0. The bug is that the functions return `pdTRUE` when it failed and `pdFALSE` when it was succesful. 
But they should return (freertos documentation):

> "pdPASS Returned only if the call to xSemaphoreTakeRecursive() was successful in obtaining the semaphore."

> "pdFAIL Returned if the call to xSemaphoreTakeRecursive() did not successfully obtain the semaphore."

> "pdPASS Returned only if the call to xSemaphoreTake() was successful in obtaining the semaphore"

> "pdFAIL Returned if the call to xSemaphoreTake() did not successfully obtain the  semaphore."


### Testing procedure


only whitelisted for `esp32-wroom-32`

To test run `BOARD=esp32-wroom-32 make -C tests/freertos/ flash test`
To see the the bug run the command before the return value fix commit (there is one for each function)
